node/task.ts

```diff
@@ -614,20 +597,38 @@ export const checkPath = im._checkPath;
  */
 export function cd(path: string): void {
     if (path) {
-        shell.cd(path);
-        _checkShell('cd');
+        let errMsg: string = '';
+        if (!fs.existsSync(path)) {
+            errMsg = loc('LIB_OperationFailed', 'cd ' + path, 'No such file or directory');
+        }
+        else if (!fs.statSync(path).isDirectory()) {
+            errMsg = loc('LIB_OperationFailed', 'cd ' + path, 'Not a directory');
+        }
+
+        if (!errMsg) {
+            process.chdir(path);
```

Comment:
Can this throw?

Reply:
Only in exceptional situations (where I think we'd want to throw). The directory is guaranteed to exist/actually be a directory so if `chdir` is throwing that indicates something weird is happening.